model/qti/metadata/guardians/MetadataGuardian.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\metadata\guardians;
+
+use core_kernel_classes_Class;
+use oat\tao\model\TaoOntology;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiItem\model\qti\metadata\MetadataValue;
+use oat\taoQtiItem\model\qti\metadata\MetadataGuardian as MetadataGuardianInterface;
+
+class MetadataGuardian extends ConfigurableService implements MetadataGuardianInterface
+{
+    public const SERVICE_ID = 'taoQtiItem/MetadataGuardian';
+
+    /** Path to find metadata value */
+    public const OPTION_EXPECTED_PATH = 'expectedPath';
+
+    /** The property URI whose value need to check */
+    public const OPTION_PROPERTY_URI = 'propertyUri';
+
+    public function guard(array $metadataValues)
+    {
+        $expectedPath = $this->getOption(self::OPTION_EXPECTED_PATH, []);
+        $propertyUri = $this->getOption(self::OPTION_PROPERTY_URI, '');
+        $class = new core_kernel_classes_Class(TaoOntology::CLASS_URI_ITEM);
```

Comment:
Is this metadata guardian guard only items?

Reply:
As I see - yes.
At the moment we have 2 metadata guardians - LomIdentifierGuardian and IbnMetadataGuardian (for ACT) - they are similar, the only difference is expected path and property to check.
And we need the same for New Meridian, that's why I made this configurable class with possibility to change expected path and property.
At the moment I didn't see any reason to make `$class` configurable.
Plus we are in `taoQtiItem` extensions, so I think it's normal, that our guardian check only items 😄